typescript/src/codeActions/custom/renameParameterToNameFromType.ts

```diff
@@ -0,0 +1,87 @@
+import { pipe, groupBy, map } from 'lodash/fp'
+import { compact } from 'lodash'
+import { CodeAction } from '../getCodeActions'
+import extractType from '../../utils/extractType'
+
+const getEdits = (
+    fileName: string,
+    position: number,
+    parameterName: string,
+    parameterIndex: number,
+    functionDeclaration: ts.Node,
+    languageService: ts.LanguageService,
+): ts.FileTextChanges[] | undefined => {
+    const typeChecker = languageService.getProgram()!.getTypeChecker()
+    const type = extractType(typeChecker, functionDeclaration)
+
+    const typeSignatureParams = typeChecker.getSignaturesOfType(type, ts.SignatureKind.Call)[0]?.parameters
+    if (!typeSignatureParams) return
+
+    const typeParamName = typeSignatureParams[parameterIndex]!.name
```

Comment:
Also sometimes it doesn't make sense to suggest to rename to internal ts name:

for example we can skip defining a name with something like this: `type Callback = ({}: {a, b}) => void` so the name would be `__0`